### PR TITLE
Ports body markings from polaris

### PR DESCRIPTION
**Pull request by Alberyk**
Ports the body marking system from polaris. And I think I got all the SQL related procs.

Todo:

- [x]  Fix the marking not showing in the mob, probably something is wrong with the organ icon creation

- [x] Look into adding some human related things and removing more dumb options

- [x]  Handle the SQL magic

### CODE REVIEW

File: code/modules/client/preference_setup/general/03_body.dm [hunk 1]
```diff
@@ -131,6 +133,8 @@ var/global/list/valid_bloodtypes = list("A+", "A-", "B+", "B-", "AB+", "AB-", "O
 			pref.organ_data = params2list(pref.organ_data)
 		if (pref.rlimb_data)
 			pref.rlimb_data = params2list(pref.rlimb_data)
+		if (pref.body_markings)
+			pref.body_markings = params2list(pref.body_markings)
```
Comment (skull132) [inline code/modules/client/preference_setup/general/03_body.dm]: For the sake of future shenanigans. Let's use `json_decode()`

File: code/modules/client/preference_setup/general/03_body.dm [hunk 2]
```diff
@@ -95,6 +99,7 @@ var/global/list/valid_bloodtypes = list("A+", "A-", "B+", "B-", "AB+", "AB-", "O
 				"disabilities" = pref.disabilities,
 				"organs_data" = list2params(pref.organ_data),
 				"organs_robotic"= list2params(pref.rlimb_data),
+				"body_markings" = json_decode(pref.body_markings),
```
Comment (skull132) [inline code/modules/client/preference_setup/general/03_body.dm]: `"body_markings" = json_encode(pref.body_markings)`

encode to turn a list into text. decode to turn text into a list. We store text on the database, we use a list ingame.

**Comment by Lohikar** [comment]
Doesn't compile.

File: code/_helpers/global_lists.dm [hunk 3]
```diff
@@ -117,6 +118,12 @@ var/global/list/cloaking_devices = list()
 			else
 				facial_hair_styles_male_list += H.name
 				facial_hair_styles_female_list += H.name
+				
+	//Body markings - Initialise all /datum/sprite_accessory/marking into an list indexed by marking name
+	paths = typesof(/datum/sprite_accessory/marking) - /datum/sprite_accessory/marking
```
Comment (Lohikar) [inline code/_helpers/global_lists.dm]: Very very minor, but `subtypesof(/datum/sprite_accessory/marking)` is a little cleaner. Functionally identical.

Comment (skull132) [inline code/_helpers/global_lists.dm]: Should clarify that `subtypesof(/datum/sprite_accessory/marking)` is the same as `typesof(/datum/sprite_accessory/marking) - /datum/sprite_accessory/marking`

And would replace the whole thing.

**Comment by skull132** [comment]
Needs changelog (plz credit Polaris in there).

**Pull request merged**